src/main/java/edu/iis/powp/events/predefine/SelectSecondTestFigureOptionListener.java

```diff
@@ -7,11 +7,32 @@
  import edu.iis.client.plottermagic.preset.FiguresJoe;
 import edu.iis.powp.app.Application;
 import edu.iis.powp.app.DriverManager;
+import edu.iis.powp.command.PlotterCommand;
+import edu.iis.powp.factory.RectangleFactory;
+import edu.iis.powp.factory.TriangleFactory;
+
  public class SelectSecondTestFigureOptionListener implements ActionListener
 {
      @Override
     public void actionPerformed(ActionEvent e)
-    {
-    	FiguresJoe.figureScript2(Application.getComponent(DriverManager.class).getCurrentPlotter());
-    }
-}
\ No newline at end of file
+    switch(e.getActionCommand())
+	{
+	case "Figure Joe 1":
```

Comment:
Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.